### ISSUE TITLE
InternalError: Packet sequence number wrong - got 1 expected 0

### ISSUE THREAD

**Issue by georgeccfisher** [PyMySQL/PyMySQL#971]
**Describe the bug**
error message appears if host is set to IP address.

**To Reproduce**
set up mariadb
```
mariadb -V
mariadb  Ver 15.1 Distrib 10.3.27-MariaDB, for debian-linux-gnueabihf (armv8l) using readline 5.2
```
set bind address to 0.0.0.0 to allow access from anywhere.
First try this:
```py
import pymysql

connection = pymysql.connect(host='localhost',
                             user='USER',
                             password='PW',
                             db='TEST'
                             )
```
-> no errors
Second try this with IP address instead of `localhost`:
```py
import pymysql

connection = pymysql.connect(host='192.168.0.42',
                             user='USER',
                             password='PW',
                             db='TEST'
                             )
```
-> error message:
```py
Traceback (most recent call last):
  File "/home/georgeccfisher/TEST.py", line 6, in <module>
    db='TEST'
  File "/usr/local/lib/python3.7/dist-packages/pymysql/connections.py", line 353, in __init__
    self.connect()
  File "/usr/local/lib/python3.7/dist-packages/pymysql/connections.py", line 632, in connect
    self._get_server_information()
  File "/usr/local/lib/python3.7/dist-packages/pymysql/connections.py", line 1055, in _get_server_information
    packet = self._read_packet()
  File "/usr/local/lib/python3.7/dist-packages/pymysql/connections.py", line 707, in _read_packet
    % (packet_number, self._next_seq_id)
pymysql.err.InternalError: Packet sequence number wrong - got 1 expected 0
```
**Expected behavior**
I would expect it to work without error message.

**Environment**
 - OS: Raspbian GNU/Linux 10 Linux version 5.10.11+ (dom@buildbot) (arm-linux-gnueabihf-gcc-8 (Ubuntu/Linaro 8.4.0-3ubuntu1) 8.4.0, GNU ld (GNU Binutils for Ubuntu) 2.34)
 - Server and version: mariadb  Ver 15.1 Distrib 10.3.27-MariaDB, for debian-linux-gnueabihf (armv8l) using readline 5.2
 - Python 3.7.3
 - PyMySQL version: 1.0.2

**Comment by methane** [comment]
Please provide a packet capture.
https://www.wireshark.org/docs/wsug_html_chunked/AppToolstcpdump.html

**Comment by georgeccfisher** [comment]
`sudo tcpdump -i any -s 65535 > packetcapture.txt`
[packetcapture.txt](https://github.com/PyMySQL/PyMySQL/files/6095086/packetcapture.txt)

**Comment by methane** [comment]
Don't use text format. Do you read the link I provide? Please use `-w <file>`.

**Comment by georgeccfisher** [comment]
`sudo tcpdump -i any -s 65535 -w packetcapture`
[packetcapture.zip](https://github.com/PyMySQL/PyMySQL/files/6095207/packetcapture.zip)

**Comment by methane** [comment]
Thank you. I confirmed your MariaDB server sent wrong packet sequence number.

![image](https://user-images.githubusercontent.com/199592/110205653-b032c200-7ebc-11eb-83fd-116d26b91273.png)

This is a MariaDB bug, not this project issue. Same issue is reported at go-sql-driver/mysql#1038. And it is reported to MariaDB already.
https://jira.mariadb.org/browse/MDEV-21252

**Comment by methane** [comment]
Although wrong packet number is MariaDB issue, "1130 Host XXX is not allowed to connect to this MariaDB server" is your environment issue.
You need to fix your MariaDB configuration.

**Comment by georgeccfisher** [comment]
Thanx. 
A solution for the MariaDB error can be found here: https://www.tecmint.com/fix-error-1130-hy000-host-not-allowed-to-connect-mysql/

**Comment by AmericanY** [comment]
> 
> 
> Thank you. I confirmed your MariaDB server sent wrong packet sequence number.
> 
> ![image](https://user-images.githubusercontent.com/199592/110205653-b032c200-7ebc-11eb-83fd-116d26b91273.png)
> 
> This is a MariaDB bug, not this project issue. Same issue is reported at [go-sql-driver/mysql#1038](https://github.com/go-sql-driver/mysql/issues/1038). And it is reported to MariaDB already.
> https://jira.mariadb.org/browse/MDEV-21252

How do you managed to get a friendly output using `tcpdump` ?

**Comment by methane** [comment]
This is wireshark. Wireshark can read pcap file created by tcpdump.

**Comment by grooverdan** [comment]
I'm sorry this too so long to fix in [MariaDB (MDEV-21252)](https://jira.mariadb.org/browse/MDEV-21252). This will be included in the next releases  10.2.42, 10.3.33, 10.4.23, 10.5.14, 10.6.6, 10.7.2.
